### ISSUE TITLE
Bump application-services  to Xcode 10 beta 6

### ISSUE THREAD

**Issue by devinreams** [mozilla-lockbox/lockbox-ios#677]
To get the latest SDK and goodies. And as soon as I do this they will drop beta 7.

- [x] contact app-services team to get a new build
- [x] update carthage
- [x] commit and push and go go go
  - [x] make sure buddybuild can build
- [x] confirm with devs they are OK with moving to beta 6

**Comment by devinreams** [comment]
Both [releases](https://github.com/mozilla/application-services/releases) 0.2.3 and publish-logins-api-1 do not currently support beta 6. I dropped a note to the app-services team...

**Comment by devinreams** [comment]
Thanks to @vladikoff we have an updated release. 😍

**Comment by vladikoff** [comment]
@devinreams try this one: https://github.com/mozilla/application-services/releases/tag/0.2.4-xcode10b6 

## Version 0.2.4

this one had a setting for Swift 4.2

```diff
diff --git a/fxa-client/sdks/swift/FxAClient/FxAClient.xcodeproj/project.pbxproj b/fxa-client/sdks/swift/FxAClient/FxAClient.xcodeproj/project.pbxproj
index 9678530..241ef0a 100644
--- a/fxa-client/sdks/swift/FxAClient/FxAClient.xcodeproj/project.pbxproj
+++ b/fxa-client/sdks/swift/FxAClient/FxAClient.xcodeproj/project.pbxproj
@@ -365,7 +365,7 @@
                                PRODUCT_NAME = "$(TARGET_NAME:c99extidentifier)";
                                SKIP_INSTALL = YES;
                                SWIFT_OPTIMIZATION_LEVEL = "-Onone";
-                               SWIFT_VERSION = 4.0;
+                               SWIFT_VERSION = 4.2;
                                TARGETED_DEVICE_FAMILY = "1,2";
                        };
                        name = Debug;
@@ -399,7 +399,7 @@
                                PRODUCT_BUNDLE_IDENTIFIER = org.mozilla.FxAClient;
                                PRODUCT_NAME = "$(TARGET_NAME:c99extidentifier)";
                                SKIP_INSTALL = YES;
-                               SWIFT_VERSION = 4.0;
+                               SWIFT_VERSION = 4.2;
                                TARGETED_DEVICE_FAMILY = "1,2";
                        };
                        name = Release;
```

**Comment by devinreams** [comment]
@vladikoff ohh it seems like its this close:👌 

```
***  Skipped installing application-services.framework binary due to the error:
	"Incompatible Swift version - framework was built with 4.2 (swiftlang-1000.0.32.1 clang-1000.10.39) and the local version is 4.2 (swiftlang-1000.0.36 clang-1000.10.44)."
```
| | swiftlang | clang |
| --- | --- | --- |
| built against | 1000.0.32.1 | 1000.10.39 | 
| expected for beta 6 |1000.0.36 | 1000.10.44 |

My local environment with beta 6 and also setting `xcode-select --switch /Applications/Xcode-beta.app/`:

```
~/src/lockbox-ios (677-bump-app-services) $ swift --version
Apple Swift version 4.2 (swiftlang-1000.0.36 clang-1000.10.44)
Target: x86_64-apple-darwin18.0.0
```

And with buddybuild set to Xcode 10 beta 6 (which seems to think its still built against 4.1.2 for no clear reason to me, I blame caches):

https://dashboard.buddybuild.com/apps/5a0ddb736e19370001034f85/build/5b913916b0fbb800014984d8#logs

Is that helpful? 😬

**Comment by vladikoff** [comment]
@devinreams try now, same tag. 

I had to ` rm -rf ~/Library/Caches/org.carthage.CarthageKit` to make sure it reclones the sources.

Seems like it worked? 

```
➜  lockbox-ios git:(677-bump-app-services) ✗ rm -rf ~/Library/Caches/org.carthage.CarthageKit
➜  lockbox-ios git:(677-bump-app-services) ✗ carthage  bootstrap                             
*** Checking out MappaMundi at "981095dff1c014a458624820ac8f2cda39c92144"
*** Checking out SwiftKeychainWrapper at "3.0.1"
*** Checking out lockbox-ios-fxa-sync at "880f310086bac8ad16c81775b3235550d9ffc7f3"
*** Checking out telemetry-ios at "v1.1.0"
*** Cloning SwiftKeychainWrapper
*** Cloning MappaMundi
*** Cloning telemetry-ios
*** Cloning lockbox-ios-fxa-sync
*** Checking out RxOptional at "3.5.0"
*** Cloning RxOptional
*** Checking out application-services at "0.2.4-xcode10b6"
*** Cloning application-services
*** Checking out Nimble at "v7.1.3"
*** Cloning Nimble
*** Checking out RxDataSources at "026857261128bf552369a3b69b3ab56c3f85704d"
*** Cloning RxDataSources
*** Checking out SwiftyJSON at "3.1.4"
*** Cloning SwiftyJSON
*** Checking out RxSwift at "4.2.0"
*** Cloning RxSwift
*** Checking out ios_sdk at "v4.14.3"
*** Cloning ios_sdk
*** Checking out sentry-cocoa at "4.1.0"
*** Cloning sentry-cocoa
*** Checking out Alamofire at "4.7.3"
*** Cloning Alamofire
*** Checking out Deferred at "35b8927c1b94ce074e10793c57e1f80d0e2227fa"
*** Cloning Deferred
*** Checking out Quick at "v1.3.1"
*** Cloning Quick
*** Checking out XCGLogger at "Version_4.0.0"
*** Cloning XCGLogger
*** xcodebuild output can be found in /var/folders/_z/8gy7l13n7yx4mdtmz4_xz_j00000gn/T/carthage-xcodebuild.CmqryQ.log
*** Downloading application-services.framework binary at "FxA SDK"
*** Downloading ios_sdk.framework binary at "Version 4.14.3"
*** Downloading RxSwift.framework binary at "Zürich"
*** Downloading sentry-cocoa.framework binary at "4.1.0"
***  Skipped installing RxSwift.framework binary due to the error:
	"Incompatible Swift version - framework was built with 4.1.2 (swiftlang-902.0.54 clang-902.0.39.2) and the local version is 4.2 (swiftlang-1000.0.36 clang-1000.10.44)."

    Falling back to building from the source
*** Building scheme "Alamofire iOS" in Alamofire.xcworkspace
*** Building scheme "Alamofire tvOS" in Alamofire.xcworkspace
```

**Comment by devinreams** [comment]
Closed at #684 (not yet on default branch, closing manually)

I also notified the team we're good for beta 6 now via Slack and GitHub 👍
